### PR TITLE
Add code of conduct, based on the Django one

### PR DESCRIPTION
**Pull request by mamoll**
### Description

This is a follow-up on the [PR I just closed](https://github.com/ros-planning/moveit/pull/1664). It adds a code of conduct to the MoveIt repo. It is based on the one used by the Django project. The previous PR was based on the Contributor Covenant / Google code of conduct. This PR hopefully better captures the spirit of what we're trying to accomplish.

### Checklist
- [x] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] Extend the tutorials / documentation [reference](http://moveit.ros.org/documentation/contributing/)
- [ ] Document API changes relevant to the user in the MIGRATION.md notes
- [ ] Create tests, which fail without this PR [reference](https://ros-planning.github.io/moveit_tutorials/)
- [ ] Include a screenshot if changing a GUI
- [ ] While waiting for someone to review your request, please help review [another open pull request](https://github.com/ros-planning/moveit/pulls) to support the maintainers

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

### CODE REVIEW

**Comment by mamoll** [comment]
@gavanderhoorn @v4hn @mlautman @davetcoleman since you commented on the other code of conduct, please let me know if this more acceptable.

File: CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+Be considerate. Your work will be used by other people, and you in turn will depend on the work of others. Any decision you take will affect users and colleagues, and you should take those consequences into account when making decisions. Remember that we're a world-wide community, so you might not be communicating in someone else's primary language.
+- **Be respectful.** Not all of us will agree all the time, but disagreement is no excuse for poor behavior and poor manners. We might all experience some frustration now and then, but we cannot allow that frustration to turn into a personal attack. It's important to remember that a community where people feel uncomfortable or threatened is not a productive one. Members of the Django community should be respectful when dealing with other members as well as with people outside the Django community.
```
Comment (j-petit) [inline CODE_OF_CONDUCT.md]:
```suggestion
- **Be respectful.** Not all of us will agree all the time, but disagreement is no excuse for poor behavior and poor manners. We might all experience some frustration now and then, but we cannot allow that frustration to turn into a personal attack. It's important to remember that a community where people feel uncomfortable or threatened is not a productive one. Members of the MoveIt community should be respectful when dealing with other members as well as with people outside the MoveIt community.
```

File: CODE_OF_CONDUCT.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  - Violent threats or language directed against another person.
+  - Discriminatory jokes and language.
+  - Posting sexually explicit or violent material.
+  - Posting (or threatening to post) other people's personally identifying information ("doxing").
+  - Personal insults, especially those using racist or sexist terms.
+  - Unwelcome sexual attention.
+  - Advocating for, or encouraging, any of the above behavior.
+  - Repeated harassment of others. In general, if someone asks you to stop, then stop.
+- **When we disagree, try to understand why.** Disagreements, both social and technical, happen all the time and Django is no exception. It is important that we resolve disagreements and differing views constructively. Remember that we're different. The strength of Django comes from its varied community, people from a wide range of backgrounds. Different people have different perspectives on issues. Being unable to understand why someone holds a viewpoint doesn't mean that they're wrong. Don't forget that it is human to err and blaming each other doesn't get us anywhere. Instead, focus on helping to resolve issues and learning from mistakes.
```
Comment (j-petit) [inline CODE_OF_CONDUCT.md]:
```suggestion
- **When we disagree, try to understand why.** Disagreements, both social and technical, happen all the time and MoveIt is no exception. It is important that we resolve disagreements and differing views constructively. Remember that we're different. The strength of MoveIt comes from its varied community, people from a wide range of backgrounds. Different people have different perspectives on issues. Being unable to understand why someone holds a viewpoint doesn't mean that they're wrong. Don't forget that it is human to err and blaming each other doesn't get us anywhere. Instead, focus on helping to resolve issues and learning from mistakes.
```

**Comment by mamoll** [comment]
> replace Django through Moveit

Oops! Thanks for fixing that!

**Comment by mamoll** [comment]
@gavanderhoorn since you felt strongest about the previous PR (PR #1664), I'd much appreciate your input on this PR, which I hope is much closer to being acceptable.

**Comment by mlautman** [comment]
This seems more positive which is nice. I would remove the list of unacceptable behaviors starting on line 16

Otherwise I'd be happy to merge

**Comment by mamoll** [comment]
@v4hn, @rhaschke What is your opinion on this?

File: CODE_OF_CONDUCT.md [hunk 3]
```diff
@@ -3,21 +3,13 @@
...
+- **Be welcoming.** We strive to be a community that welcomes and supports people of all backgrounds and identities. 
+- **Be considerate.** Your work will be used by other people, and you in turn will depend on the work of others. Any decision you take will affect users and colleagues, and you should take those consequences into account when making decisions. Remember that we're a world-wide community, so you might not be communicating in someone else's primary language.
 - **Be respectful.** Not all of us will agree all the time, but disagreement is no excuse for poor behavior and poor manners. We might all experience some frustration now and then, but we cannot allow that frustration to turn into a personal attack. It's important to remember that a community where people feel uncomfortable or threatened is not a productive one. Members of the MoveIt community should be respectful when dealing with other members as well as with people outside the MoveIt community.
```
Comment (rhaschke) [inline CODE_OF_CONDUCT.md]: Can we merge this paragraph with the one below, "when we disagree, try to understand why"? I think, both paragraphs point in the same direction. Merging them, we could get rid of some negative tone here as well - again emphasizing the positive, desired behavior.

**Comment by rhaschke** [comment]
@mamoll, thanks for iterating on this. I think it reads much more positive already.

File: CODE_OF_CONDUCT.md (hunk 3 above)
Comment (mamoll) [reply CODE_OF_CONDUCT.md]: To me, they are subtly different. This paragraph encourages the reader to be mindful of their own behavior, while the other focuses on trying to understand the behavior of others.

Comment (rhaschke) [reply CODE_OF_CONDUCT.md]: In my point of view, a large part of this paragraph deals with how to handle disagreement. Maybe just move (or remove) this part here, as this is handled in the other paragraph as well?

**Comment by mamoll** [comment]
All feedback has been considered. Merging as is. I'm very open to reviewing additional changes in a follow on PR

**Comment by davetcoleman** [comment]
Thanks @mamoll !

**Pull request merged**